src/views/Reticket/Confirm.js

```diff
@@ -0,0 +1,63 @@
+import React, { useState } from 'react';
+import * as need from 'lib/need';
+import { Just, Nothing } from 'folktale/maybe';
+
+import PaperCollateralRenderer from 'PaperCollateralRenderer';
+
+import { usePointCursor } from 'store/pointCursor';
+
+import View from 'components/View';
+import { ForwardButton } from 'components/Buttons';
+
+import useLifecycle from 'lib/useLifecycle';
+import { generateWallet } from 'lib/invite';
+import { useLocalRouter } from 'lib/LocalRouter';
+
+import { isDevelopment } from 'lib/flags';
+
+export default function Confirm({ STEP_NAMES, newWallet, storeNewWallet }) {
+  const { push } = useLocalRouter();
+  const { pointCursor } = usePointCursor();
+  const point = need.point(pointCursor);
+
+  const [generatedWallet, setGeneratedWallet] = useState(Nothing());
+
+  // start generating new wallet on mount
+  //TODO also check we can pay for the transactions?
+  useLifecycle(() => {
+    generateWallet(point).then(wal => {
+      if (isDevelopment) {
+        console.log('ticket for', point, wal.ticket);
+      }
+      setGeneratedWallet(Just(wal));
+    });
+  });
+
+  const next = () => {
+    push(STEP_NAMES.DOWNLOAD);
+  };
+
+  const paperRenderer = Nothing.hasInstance(generatedWallet) ? null : (
+    <PaperCollateralRenderer
+      wallet={{ [point]: generatedWallet.value }}
+      className={'extremely-hidden'}
```

Comment:
same notes from paper collateral renderer above

Reply:
actually, can we abstract this even more somehow, since these code blocks are basically identical?